### PR TITLE
Add type support for ignition tests using hardhat-viem (VIEM 2)

### PR DESCRIPTION
**Pull request by zoeyTM**
Note: currently this only supports types for local contracts. Support for imported contracts and contracts with external artifacts will come after an update to `hardhat-viem`

### CODE REVIEW

File: packages/hardhat-plugin-viem/src/viem-ignition-helper.ts [hunk 1]
```diff
@@ -49,54 +57,29 @@ export class IgnitionHelper {
...
-      throw new HardhatPluginError("hardhat-ignition-viem", message);
-    }
-
-    return this._toViemContracts(ignitionModule, result);
+      successfulDeploymentResult
+    );
   }
 
-  private async _toViemContracts<
+  private static async _toViemContracts<
```
Comment (zoeyTM) [inline packages/hardhat-plugin-viem/src/viem-ignition-helper.ts:77]: What's the reasoning behind making this static but not the ethers equivalent functions?

File: packages/hardhat-plugin-ethers/src/index.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ extendEnvironment((hre) => {
   if (
     hre.ignition !== undefined &&
     hre.ignition.type !== "stub" &&
-    hre.ignition.type !== "ethers"
+    hre.ignition.type !== "viem"
```
[the remark this answers is not in the document]
Comment (kanej) [reply packages/hardhat-plugin-ethers/src/index.ts]: Agreed. Good catch, clearly I didn't think that through.

File: packages/hardhat-plugin-viem/src/viem-ignition-helper.ts (hunk 1 above)
Comment (kanej) [reply packages/hardhat-plugin-viem/src/viem-ignition-helper.ts:77]: Agreed. I will bring the ethers version inline.

**Review by kanej — APPROVED** [review]
I am going to merge this into the base viem branch, so it can all be reviewed in one place.

**Pull request merged**